### PR TITLE
SW-5564 Associate variable values with projects

### PR DESCRIPTION
**Pull request by sgrimm**
We need to support variable values that are referenced by different types of
documents and variable values that don't have any associated documents at all
(questionnaire deliverable answers).

Modify the data model such that variable values are associated with projects
instead of with documents.

This change does not modify the existing API, which continues to address values
using document IDs.

With this change, it becomes impossible to have different values for the same
variable in different documents for the same project, which effectively means
there can only be one document from each template per project (it's fine to
have documents of different templates since each template will have separate
section variables). This limitation has been okayed by the product team.

### CODE REVIEW

**Comment by sgrimm** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2075?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2075** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2075?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2033** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2033?utm_source=stack-comment-icon" target="_blank">Graphite</a>: 14 other dependent PRs ([#2029](https://github.com/terraware/terraware-server/pull/2029) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2029?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2064](https://github.com/terraware/terraware-server/pull/2064) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2064?utm_source=stack-comment-icon" target="_blank">Graphite</a>, [#2065](https://github.com/terraware/terraware-server/pull/2065) <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/2065?utm_source=stack-comment-icon" target="_blank">Graphite</a> and 11 others)
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @sgrimm and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/main/resources/db/migration/0250/V282__DocumentProducerTables.sql [hunk 1]
```diff
@@ -197,7 +197,7 @@ CREATE TABLE docprod.document_saved_versions (
 
 CREATE TABLE docprod.variable_values (
     id BIGINT PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY,
-    document_id BIGINT NOT NULL REFERENCES docprod.documents ON DELETE CASCADE,
+    project_id BIGINT NOT NULL REFERENCES projects,
```
Comment (tommylau523) [inline src/main/resources/db/migration/0250/V282__DocumentProducerTables.sql]: Should we have ON DELETE CASCADE for if a project gets deleted? 

Which actually prompts me to think about, is there any thing to stop a user from deleting an accelerator project?

**Review by tommylau523 — APPROVED** [review]

File: src/main/resources/db/migration/0250/V282__DocumentProducerTables.sql (hunk 1 above)
Comment (sgrimm) [reply src/main/resources/db/migration/0250/V282__DocumentProducerTables.sql]: Ah, good call, not sure why I left that off here.

Currently you can't delete a project if it has accelerator details (there's no ON DELETE on the details table), though honestly I don't remember if that was intentional or an oversight.

**Review by nickgraz — APPROVED** [review]
This all looks good to me, assuming that we don't need to support multiple documents.

**Pull request merged**